openedx/features/course_experience/views/course_home_messages.py

```diff
@@ -107,36 +110,56 @@ def _register_course_home_messages(request, course, user_access, course_start_da
     """
     Register messages to be shown in the course home content page.
     """
-    if user_access['is_anonymous']:
-        CourseHomeMessages.register_info_message(
-            request,
-            Text(_(
-                '{sign_in_link} or {register_link} and then enroll in this course.'
-            )).format(
-                sign_in_link=HTML('<a href="/login?next={current_url}">{sign_in_label}</a>').format(
-                    sign_in_label=_('Sign in'),
-                    current_url=urlquote_plus(request.path),
+
+    is_course_public = allow_public_access(course, [COURSE_VISIBILITY_PUBLIC])
+    is_self_enrollment_allowed = course_open_for_self_enrollment(course.id)
+
+    if is_course_public and is_self_enrollment_allowed:
+        if not user_access['is_anonymous'] and not user_access['is_staff'] and not user_access['is_enrolled']:
+            CourseHomeMessages.register_info_message(
+                request,
+                Text(_(
+                    '{open_enroll_link}Enroll now{close_enroll_link} to access the full course.'
+                )).format(
+                    open_enroll_link=HTML('<button class="enroll-btn btn-link">'),
+                    close_enroll_link=HTML('</button>')
                 ),
-                register_link=HTML('<a href="/register?next={current_url}">{register_label}</a>').format(
-                    register_label=_('register'),
-                    current_url=urlquote_plus(request.path),
+                title=Text(_('Welcome to {course_display_name}')).format(
+                    course_display_name=course.display_name
+                )
+            )
+
+    if not is_course_public:
+        if user_access['is_anonymous']:
```

Comment:
Nit: You can probably combine these two conditions like `if user_access['is_anonymous'] and not is_course_public:`